### DISCUSSION TITLE
ReverseMap equivalent for CreateProjection

### DISCUSSION THREAD

**Discussion by jeffmccormick** [LuckyPennySoftware/AutoMapper#3953] [Ideas]
### Question
With the new projection API, are there plans to include a `ReverseMap()` equivalent for `IProjectionExpression<>`? Or is the recommendation to simply create a separate map for the reverse direction.

The [upgrade guide](https://docs.automapper.org/en/latest/11.0-Upgrade-Guide.html#createprojection) does not mention a path to follow for reverse mapping. It seems plausible that a `IProjectionExpression<>` could be reversed into an `IMappingExpression<>`.

### Example

```csharp
CreateProjection<Foo, Bar>()
    .ForMember(d => d.Name, opt => opt.MapFrom(src => src.Thing.Name))
    .ReverseMap() // Does not currently exist, would transition to IMappingExpression
    // The reverse map would perform the following line for us
    .ForPath(d => d.Thing.Name, opt => opt.MapFrom(src => src.Name));
```

**Comment by jbogard** [comment]
Nope I don't think so. I don't see why you'd project in two directions?

**Comment by jbogard** [comment]
Ahhh you want to project in one direction, map in the other, I see now.

**Reply by jbogard** [reply]
I'm not sure there's a technical reason why this wouldn't work - the interfaces share a common base class. We just have the interface to limit the configuration to what is possible for projections.

**Reply by lbargaoanu** [reply]
The only reason is to keep a clean API for `CreateProjection`, which is, after all, just a query. So a projection could support only the main use cases for a query. The rest can be done with `CreateMap` or the `Advanced()` extension method.

**Reply by jeffmccormick** [reply]
The projection is just a query, but I think this is a relatively common use case to want to map back to the entity. That being said, the clean API is an acceptable argument since it is easy enough to create a separate map. There may be some code duplication, but likely minimal in the common case. This can be marked as answered. Thanks!

**Comment by lbargaoanu** [comment]
If you must, see [this](https://github.com/AutoMapper/AutoMapper/pull/3920/commits/58410aba51033f8e189a0053602908da35fea34f). But I would simply create the map explicitly.
